src/validation/validateFencesExistence.ts

```diff
@@ -0,0 +1,72 @@
+import reportError from '../core/reportError';
+import * as glob from 'glob';
+import * as path from 'path';
+import getAllConfigs from '../utils/getAllConfigs';
+import Config from '../types/Config';
+import ConfigSet from '../types/ConfigSet';
+import dedupe from '../utils/dedupe';
+
+interface ConfigPathPair {
+    path: string;
+    config: Config;
+}
+
+export default function validateFencesExistence() {
+    const configPathMap = getAllConfigs();
+    const requiredFenceGlobs = getRequiredFenceGlobs(configPathMap);
+
+    // get a deduped list of required paths
+    const requiredPaths = dedupe(requiredFenceGlobs.map(f => glob.sync(f)));
+
+    // get the required paths that aren't in our map of all configs
+    const missingRequiredPaths = requiredPaths.filter(p => !configPathMap[p]);
+
+    if (missingRequiredPaths.length > 0) {
+        missingRequiredPaths.forEach(p => reportError(`Missing fence.json at ${p}`));
```

Comment:
>reportError(`Missing fence.json at ${p}`) [](start = 42, length = 41)

Looks like this conflicts with my error reporting revamp.  This doesn't fit the typical pattern for a validation error, but we should be able to figure out some uniform format for errors.